src/blueprints/tv2/action-factories/tv2-action-factory-provider.ts

```diff
@@ -0,0 +1,328 @@
+import { AtemToTriCasterSplitScreenConverter } from '../helpers/atem-to-tricaster-split-screen-converter'
+import { Tv2MisconfigurationException } from '../exceptions/tv2-misconfiguration-exception'
+import {
+  Tv2SisyfosAudioTimelineObjectFactory
+} from '../timeline-object-factories/tv2-sisyfos-audio-timeline-object-factory'
+import {
+  Tv2VideoClipTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-video-clip-timeline-object-factory'
+import { Tv2AssetPathHelper } from '../helpers/tv2-asset-path-helper'
+import { Tv2CasparCgTimelineObjectFactory } from '../timeline-object-factories/tv2-caspar-cg-timeline-object-factory'
+import { Tv2ActionManifestMapper } from '../helpers/tv2-action-manifest-mapper'
+import {
+  Tv2GraphicsTimelineObjectFactoryFactory
+} from '../timeline-object-factories/tv2-graphics-timeline-object-factory-factory'
+import { Tv2StringHashConverter } from '../helpers/tv2-string-hash-converter'
+import {
+  Tv2GraphicsSplitScreenTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-graphics-split-screen-timeline-object-factory'
+import {
+  Tv2RobotTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-robot-timeline-object-factory'
+import {
+  Tv2TelemetricsTimelineObjectFactory
+} from '../timeline-object-factories/tv2-telemetrics-timeline-object-factory'
+
+
+interface ActionFactoryInstance<T> {
+  factory: T,
+  shouldFactoryBeRecreated: (configuration: Tv2BlueprintConfiguration) => boolean
+}
+
+export class Tv2ActionFactoryProvider {
+
+  private cameraActionFactoryInstance: ActionFactoryInstance<Tv2CameraActionFactory>
+  private remoteActionFactoryInstance: ActionFactoryInstance<Tv2RemoteActionFactory>
+  private transitionEffectActionFactoryInstance: ActionFactoryInstance<Tv2TransitionEffectActionFactory>
+  private audioActionFactoryInstance: ActionFactoryInstance<Tv2AudioActionFactory>
+  private graphicsActionFactoryInstance: ActionFactoryInstance<Tv2GraphicsActionFactory>
+  private videoClipActionFactoryInstance: ActionFactoryInstance<Tv2VideoClipActionFactory>
+  private videoMixerConfigurationActionFactoryInstance: ActionFactoryInstance<Tv2VideoMixerConfigurationActionFactory>
+  private splitScreenActionFactoryInstance: ActionFactoryInstance<Tv2SplitScreenActionFactory>
+  private replayActionFactoryInstance: ActionFactoryInstance<Tv2ReplayActionFactory>
+  private robotActionFactoryInstance: ActionFactoryInstance<Tv2RobotActionFactory>
+
+  public createCameraActionFactory(configuration: Tv2BlueprintConfiguration): Tv2CameraActionFactory {
+    this.cameraActionFactoryInstance = this.getUpdatedActionFactoryInstance(
+      this.cameraActionFactoryInstance,
+      () => {
+        return new Tv2CameraActionFactory(
+          this.createVideoMixerTimelineObjectFactory(configuration),
+          this.createAudioTimelineObjectFactory()
+        )
+      },
+      (c: Tv2BlueprintConfiguration): boolean => {
```

Comment:
Is this the case where we would be alright with just using `c` since we refer to a configuration in the context where we already have a configuration?

Reply:
`c` isn't ideal. It's not uncommon though to name variables in arrow functions like these and as you point out, we are doing it here since we already have a variable called `configuration` within the scope.

The only other names I can come up with is to name `c` `oldConfiguration` and `configuration` `newConfiguration` which I am also not that much of a fan of...

Reply:
Isn't it `configuration` that is the old configuration? (except for first call where it acts as both.

And personally I prefer `newConfiguration` over `c`, e.g. due to the potential confusion of what is the original and what is the changed :)

Reply:
No, `configuration` is the new configuration since that is the newest configuration provided when the `createXActionFactory()` is being called.